# HG changeset patch
# User Ken Brown <kbrown@cornell.edu>
# Date 1426247343 14400
#      Fri Mar 13 07:49:03 2015 -0400
# Node ID e2f47d3ad784831ace3ab5d4021e5996cf7e1877
# Parent  74b47369b904fba3b89b479173268d5382ab7189
* src/clisp-link.in (lncp): Don't symlink a DLL on Cygwin

diff -r 74b47369b904 -r e2f47d3ad784 src/clisp-link.in
--- a/src/clisp-link.in	Thu Mar 05 12:30:28 2015 -0500
+++ b/src/clisp-link.in	Fri Mar 13 07:49:03 2015 -0400
@@ -23,12 +23,20 @@
   echo `cd $1; cd ${dirname}; pwd`/`basename "$2"`
 }
 
+HSYSOS="@HSYSOS@"
+
 lncp () {
-  # Make a link from $1 to $2. Try symbolic link, hard link, file copying.
+  # Make a link from $1 to $2.  Try symbolic link, hard link, file
+  # copying.  Don't use a link on Cygwin if $1 is a DLL, because the
+  # Windows loader doesn't understand Cygwin symlinks.
   rm -f "$2"
-  ln -s "`normalize . $1`" "$2" 2>/dev/null || \
-    ln "$1" "$2" 2>/dev/null || \
+  if [ "${HSYSOS}" = cygwin -a ${1/.dll} != $1 ] ; then 
     cp -p "$1" "$2"
+  else
+    ln -s "`normalize . $1`" "$2" 2>/dev/null || \
+      ln "$1" "$2" 2>/dev/null || \
+      cp -p "$1" "$2"
+  fi
 }
 
 lncp_some () {
